services/index/src/data.js

```diff
@@ -1,198 +1,623 @@
-let _ = require('lodash');
-let Entity = require('taskcluster-lib-entities');
-
-/** Entities for indexed tasks */
-let IndexedTask = Entity.configure({
-  version: 1,
-  partitionKey: Entity.keys.HashKey('namespace'),
-  rowKey: Entity.keys.StringKey('name'),
-  properties: {
-    namespace: Entity.types.String,
-    name: Entity.types.String,
-    rank: Entity.types.Number,
-    taskId: Entity.types.SlugId,
-    data: Entity.types.JSON,
-    expires: Entity.types.Date,
-  },
-});
+const _ = require('lodash');
+const assert = require('assert').strict;
+const taskcluster = require('taskcluster-client');
+const {UNIQUE_VIOLATION} = require('taskcluster-lib-postgres');
+const helpers = require('./helpers');
+const { paginateResults } = require('taskcluster-lib-api');
 
-// Export IndexedTask
-exports.IndexedTask = IndexedTask;
-
-/** Get JSON representation of indexed task */
-IndexedTask.prototype.json = function() {
-  let ns = this.namespace + '.' + this.name;
-  // Remove separate if there is no need
-  if (this.namespace.length === 0 || this.name.length === 0) {
-    ns = this.namespace + this.name;
-  }
-  return {
-    namespace: ns,
-    taskId: this.taskId,
-    rank: this.rank,
-    data: _.cloneDeep(this.data),
-    expires: this.expires.toJSON(),
-  };
+const makeError = (message, code, statusCode) => {
+  const err = new Error(message);
+  err.code = code;
+  err.name = `${code}Error`;
+  err.statusCode = statusCode;
+  return err;
 };
 
-/** Entities for namespaces */
-let Namespace = Entity.configure({
-  version: 1,
-  partitionKey: Entity.keys.HashKey('parent'),
-  rowKey: Entity.keys.StringKey('name'),
-  properties: {
-    parent: Entity.types.String,
-    name: Entity.types.String,
-    expires: Entity.types.Date,
-  },
-});
+const make404 = () => makeError('Resource not found', 'ResourceNotFound', 404);
 
-// Export Namespace
-exports.Namespace = Namespace;
+class IndexedTask {
```

Comment:
As I understand it we only need these classes if they're getting passed around a lot and not just CRUD'ed inside api endpoints. Could we get away with not having these in index?

Reply:
Done.